sematic/plugins/storage/tests/test_local_storage.py

```diff
@@ -0,0 +1,34 @@
+# Standard Library
+import tempfile
+from unittest import mock
+
+# Third-party
+import flask.testing
+
+# Sematic
+from sematic.api.tests.fixtures import mock_auth, test_client  # noqa: F401
+from sematic.db.tests.fixtures import test_db  # noqa: F401
+from sematic.plugins.storage.local_storage import LocalStorage
+
+
+def test_upload_download(
+    mock_auth,  # noqa: F811
+    test_client: flask.testing.FlaskClient,  # noqa: F811
+):
+    value = b"foo"
+
+    local_storage = LocalStorage()
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        with mock.patch(
+            "sematic.config.config_dir.get_config_dir", return_value=tempdir
+        ):
+            response = test_client.put(
+                local_storage.get_write_location("artifacts", "123"), data=value
+            )
+
+            assert response.status_code == 200
```

Comment:
The implementation looks great! 

Can you help add an assertion that the returned URL is indeed the upload endpoint? 

Ditto for the memory storage.  
//------------------------------
Also, is it possible to test the upload endpoint as well? Say, can we mock the file stream API, and see the request payload is indeed written to a file? 

Considering the PR is too large, if you think it's doable, feel free to do the complete test in a separate PR. Thanks!